### PR TITLE
Add command

### PR DESCRIPTION
**Pull request by marcustw**
Completed add command with no address

### CODE REVIEW

File: src/main/resources/view/PersonListCard.fxml [hunk 1]
```diff
@@ -29,8 +29,8 @@
       </HBox>
       <FlowPane fx:id="tags" />
       <Label fx:id="phone" styleClass="cell_small_label" text="\$phone" />
-      <Label fx:id="address" styleClass="cell_small_label" text="\$address" />
       <Label fx:id="email" styleClass="cell_small_label" text="\$email" />
+      <Label fx:id="matriculationNumber" styleClass="cell_small_label" text="\$email" />
```
Comment (erisjacey) [inline src/main/resources/view/PersonListCard.fxml]: Shouldn't the text name for this be `matriculationNumber`?

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 2]
```diff
@@ -43,15 +42,15 @@
...
+    public static final String MATRIC_NO_AMY = " " + PREFIX_MATRICULATION_NUMBER + VALID_MATRIC_NO_AMY;
+    public static final String MATRIC_NO_BOB = " " + PREFIX_MATRICULATION_NUMBER + VALID_MATRIC_NO_BOB;
     public static final String TAG_DESC_FRIEND = " " + PREFIX_TAG + VALID_TAG_FRIEND;
     public static final String TAG_DESC_HUSBAND = " " + PREFIX_TAG + VALID_TAG_HUSBAND;
 
     public static final String INVALID_NAME_DESC = " " + PREFIX_NAME + "James&"; // '&' not allowed in names
     public static final String INVALID_PHONE_DESC = " " + PREFIX_PHONE + "911a"; // 'a' not allowed in phones
     public static final String INVALID_EMAIL_DESC = " " + PREFIX_EMAIL + "bob!yahoo"; // missing '@' symbol
-    public static final String INVALID_ADDRESS_DESC = " " + PREFIX_ADDRESS; // empty string not allowed for addresses
+    public static final String INVALID_MATRIC_NO = " " + PREFIX_MATRICULATION_NUMBER + "A123B"; //3 digits
```
Comment (erisjacey) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:53]: Maybe `INVALID_MATRIC_NO` could be changed to `INVALID_MATRIC_NUM` instead?

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 3]
```diff
@@ -62,10 +61,10 @@
 
     static {
         DESC_AMY = new EditPersonDescriptorBuilder().withName(VALID_NAME_AMY)
-                .withPhone(VALID_PHONE_AMY).withEmail(VALID_EMAIL_AMY).withAddress(VALID_ADDRESS_AMY)
+                .withPhone(VALID_PHONE_AMY).withEmail(VALID_EMAIL_AMY)
```
Comment (erisjacey) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:64]: Shouldn't there be a method call `withMatriculationNumber(...)` as well?

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
-import static seedu.address.logic.commands.CommandTestUtil.ADDRESS_DESC_BOB;
 import static seedu.address.logic.commands.CommandTestUtil.EMAIL_DESC_AMY;
 import static seedu.address.logic.commands.CommandTestUtil.EMAIL_DESC_BOB;
-import static seedu.address.logic.commands.CommandTestUtil.INVALID_ADDRESS_DESC;
 import static seedu.address.logic.commands.CommandTestUtil.INVALID_EMAIL_DESC;
 import static seedu.address.logic.commands.CommandTestUtil.INVALID_NAME_DESC;
 import static seedu.address.logic.commands.CommandTestUtil.INVALID_PHONE_DESC;
 import static seedu.address.logic.commands.CommandTestUtil.INVALID_TAG_DESC;
+import static seedu.address.logic.commands.CommandTestUtil.MATRIC_NO_AMY;
+import static seedu.address.logic.commands.CommandTestUtil.MATRIC_NO_BOB;
```
Comment (erisjacey) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: Perhaps `MATRIC_NO_*` could be changed to `MATRIC_NUM_*`?

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 5]
```diff
@@ -48,37 +46,39 @@ public void parse_allFieldsPresent_success() {
...
+                + MATRIC_NO_BOB
+                + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
 
         // multiple tags - all accepted
         Person expectedPersonMultipleTags = new PersonBuilder(BOB).withTags(VALID_TAG_FRIEND, VALID_TAG_HUSBAND)
                 .build();
-        assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
+        assertParseSuccess(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB
+                + MATRIC_NO_BOB
                 + TAG_DESC_HUSBAND + TAG_DESC_FRIEND, new AddCommand(expectedPersonMultipleTags));
```
Comment (erisjacey) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: Should there be an awkward line break right after `MATRIC_NO_BOB`? Same issue with the other test cases.

File: src/test/java/seedu/address/model/person/NameContainsKeywordsPredicateTest.java [hunk 6]
```diff
@@ -70,6 +70,6 @@ public void test_nameDoesNotContainKeywords_returnsFalse() {
         // Keywords match phone, email and address, but does not match name
         predicate = new NameContainsKeywordsPredicate(Arrays.asList("12345", "alice@email.com", "Main", "Street"));
         assertFalse(predicate.test(new PersonBuilder().withName("Alice").withPhone("12345")
-                .withEmail("alice@email.com").withAddress("Main Street").build()));
+                .withEmail("alice@email.com").build()));
```
Comment (erisjacey) [inline src/test/java/seedu/address/model/person/NameContainsKeywordsPredicateTest.java]: Shouldn't there be a method call `withMatriculationNumber(...)`?

**Review by erisjacey — COMMENTED** [review]
LGTM. Just a few nitpicks 😄

File: src/test/java/seedu/address/testutil/PersonBuilder.java [hunk 7]
```diff
@@ -89,8 +81,16 @@ public PersonBuilder withEmail(String email) {
         return this;
     }
 
+    /**
+     * Sets the {@code Email} of the {@code Person} that we are building.
```
Comment (nweiyue) [inline src/test/java/seedu/address/testutil/PersonBuilder.java]: Should this be "{@ MatriculationNumber}" instead of "Email"?

**Review by nweiyue — COMMENTED** [review]
LGTM. Just a nitpick.

File: src/main/resources/view/PersonListCard.fxml (hunk 1 above)
Comment (marcustw) [reply src/main/resources/view/PersonListCard.fxml]: Ahhh I see, thanks!

File: src/test/java/seedu/address/model/person/NameContainsKeywordsPredicateTest.java (hunk 6 above)
Comment (marcustw) [reply src/test/java/seedu/address/model/person/NameContainsKeywordsPredicateTest.java]: Yes there should be. Missed it out.

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java (hunk 5 above)
Comment (marcustw) [reply src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: This was left with the intention of easy deleting if there comes an error.

File: src/test/java/seedu/address/testutil/PersonBuilder.java (hunk 7 above)
Comment (marcustw) [reply src/test/java/seedu/address/testutil/PersonBuilder.java]: Ahh I see. Thanks!

**Comment by marcustw** [comment]
Changed according to comments

**Comment by marcustw** [comment]
Went ahead with Sheng Yang's version in PR #64 .

**Pull request closed without merging**